Applications/WebPortal/WebApp/library/source/pages/loading_page.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import { Transition } from 'react-transition-group';
+import { HBoxLayout, Padding, VBoxLayout } from '..';
+
+interface Properties {
+
+  /** Ammount of time in milliseconds before
+   * displaying the loading animation. */
+  delay?: number;
+}
+
+interface State {
+  showAnimation: boolean;
+}
+
+/** Displays a loading page. */
+export class LoadingPage extends React.Component<Properties, State> {
+  public static readonly defaultProps = {
+    delay: 0
+  }
+
+  constructor(props: Properties) {
+    super(props);
+    this.state = {
+      showAnimation: false
+    };
+  }
+
+  public render(): JSX.Element {
+    return (
+      <VBoxLayout width='100%' height='100%'>
+        <Padding size={LoadingPage.TOP_PADDING}/>
+        <HBoxLayout>
+          <Padding/>
+          <Transition in={this.state.showAnimation} timeout={this.props.delay}>
+            {(state) => (
+              <div style={{ ...(LoadingPage.DELAY as any)[state] }}>
+                <img src='resources/loading/pre-loader.gif'
+                  style={LoadingPage.STYLE.containerStyle}/>
+              </div>)}
+          </Transition>
+          <Padding/>
+        </HBoxLayout>
+        <Padding size={LoadingPage.BOTTOM_PADDING}/>
+      </VBoxLayout>);
+  }
+
+  public componentDidMount(): void {
+    this.setState({showAnimation: true});
```

Comment:
Does this state do anything? What happens if you remove this state and pass a value of ```true``` to the ```Transition```'s ```in``` prop?

Reply:
I originally tried it with the `in` prop being set to `true` and the transition would not occur.

Reply:
Okay, keep it as is then.